src/js/components/DataTable/index.d.ts

```diff
@@ -102,8 +105,14 @@ export interface DataTableProps<TRowType = any> {
   onClickRow?: (event: MouseClick<TRowType> | KeyPress<TRowType>) => void;
   onMore?: () => void;
   onSearch?: (search: string) => void;
-  onSelect?: (select: (string | number)[]) => void;
+  onSelect?: (select: (string | number)[], datum: TRowType) => void;
   onSort?: (sort: { property: string; direction: 'asc' | 'desc' }) => void;
+  onUpdate?: (datatableState: {
+    sort: { property: string; direction: 'asc' | 'desc' };
```

Comment:
Can we make `sort` and `expanded` optional, an only include them if the properties to enable them are present we we call `onUpdate`? I'm wanting to keep the interface simpler for the typical case.

Reply:
Certainly. Is it OK if the expand/sort member is in the object but is undefined or just shouldn't be there? A little messier code to not include it at all.